### PR TITLE
Week 4 Day 7 PWA UI UX

### PR DESCRIPTION
**Pull request by ststimac**

### CODE REVIEW

File: docs/30DaysOfPWA/platforms-practices/07.md [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
 
+Welcome to day 29 of the [30 Days of PWA](<https://aka.ms/learn-pwa/30Days-blog>) series! Today’s blog post will walk through some of the best practices to keep in mind when thinking about your PWA’s user experience and interface.  
 
-| Section | Description |
-| ------- | ----------- |
-| **Overview** |How to maintain your PWAs responsiveness across different platforms? How to keep your PWAs content focused?|
-| **Motivation** | Maintaining a consistent native user experience is key to a good PWA. Seamless user interactions for your PWA. |
-| **Example** | Design example. |
+## Responsive considerations 
+When designing your PWA there are a few things to think about beyond just ensuring the user interface adapts to whatever device someone is on, whether that’s on mobile, a tablet or a desktop. In a previous article we looked at [display modes](<https://microsoft.github.io/win-student-devs/#/30DaysOfPWA/advanced-capabilities/04>) and how you can make your PWA feel more like an app. These display modes are important to consider when you’re thinking about the responsiveness of your PWA.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/platforms-practices/07.md:9]: Simplifying the link

```suggestion
When designing your PWA there are a few things to think about beyond just ensuring the user interface adapts to whatever device someone is on, whether that’s on mobile, a tablet or a desktop. In a previous article we looked at [display modes](../advanced-capabilities/04.md) and how you can make your PWA feel more like an app. These display modes are important to consider when you’re thinking about the responsiveness of your PWA. 
```

File: docs/30DaysOfPWA/platforms-practices/07.md [hunk 2]
```diff
@@ -1,13 +1,44 @@
...
-| **Example** | Design example. |
+## Responsive considerations 
+When designing your PWA there are a few things to think about beyond just ensuring the user interface adapts to whatever device someone is on, whether that’s on mobile, a tablet or a desktop. In a previous article we looked at [display modes](<https://microsoft.github.io/win-student-devs/#/30DaysOfPWA/advanced-capabilities/04>) and how you can make your PWA feel more like an app. These display modes are important to consider when you’re thinking about the responsiveness of your PWA. 
+
+When a user is on mobile or a tablet, how much of the browser UI do you want to be shown, if any? Does your decision affect the UI of the PWA? For example, if you want your PWA to be displayed in fullscreen mode, ensure there’s a way to navigate back from the current page because there’s no browser back button provided in this mode.
+
+How do you want your desktop version to display if it’s in standalone mode outside of the browser? Do you want it to feel as native as possible? Consider implementing Window Controls Overlay to customize the title bar of your PWA.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/platforms-practices/07.md:13]: Consider linking "Windows Controls Overlay" to the relevant section in the display modes article.

File: docs/30DaysOfPWA/platforms-practices/07.md [hunk 3]
```diff
@@ -1,13 +1,44 @@
...
+
+![The Starbucks PWA home screen and menu](_media/strabucks-app.jpg)
+
+The rest of the site content is deprioritized and placed at the bottom. It’s still accessible but out of the way of the main task users are coming to the site for: to order coffee. 
+
+Identify what key tasks and content your users are coming to your app for and optimize for those scenarios to streamline their task. Focus on reducing any clutter in the UI particularly with mobile versions due to the reduced screen size. For example, the Twitter app on desktop has two sidebars. On the mobile version, the navigation moves to the bottom and the “What’s Happening” sidebar becomes integrated into the search tab with the main focus of the app being the timeline.  
+
+Additionally consider how to prompt your users to install the latest version of your PWA when there’s been a change, especially if it’s going to affect what they’re seeing in the app. Make it stand out from the rest of the UI so that users notice the call to update immediately so that they can start their journey in your app with the most recent version and no interruption to update in the middle of using it. 
+
+## Keep users engaged in offline mode
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/platforms-practices/07.md:28]: Can we link this section to a few other places in the blog series? I think we have some cool content about offline support we could link to. Any page in the series could be "page 1" for somebody, so I feel it's important to offer links throughout the series.

File: docs/30DaysOfPWA/platforms-practices/07.md [hunk 4]
```diff
@@ -1,13 +1,44 @@
...
+
+Having a typical “page offline” or “no connection” message won’t keep your users engaged and in the app, and it doesn’t take advantage of the offline capabilities of PWAs. Think about what the experience will be offline. What content can still be served and interacted with when offline, and if there isn’t core content that can be, what can you put in its place that isn’t an error message?
+
+## Make interactions feel seamless
+Performance and speed are critical when thinking about the user experience of any website. If you’re on a website and click something and the navigation takes you to a screen that is loading slowly, it can be frustrating especially without any indication the page is loading. Provide transition states such as showing skeleton UI when content is loading. This provides a preview of the UI while it waits for the actual UI to load. It also indicates that something is happening and provides a better experience than just a loader or progress bar. The switch from skeleton UI to actual UI is smoother and sets your users up for what to expect on the page.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/platforms-practices/07.md:34]: Can we either define what skeleton UI means somewhere here, or link to a page that does? I'm afraid not everyone starting on their dev journey knows what it is.

File: docs/30DaysOfPWA/platforms-practices/07.md [hunk 5]
```diff
@@ -1,13 +1,44 @@
...
+
+Seamlessness and perceived performance aren’t just about the UI of your PWA though, certain tasks that users need to complete can be sped up with certain features. For example, in a checkout task flow, make things like autofill and web payments available so that users can quickly fill out details. This reduces friction and gets them to the end of their task more quickly. Experiences like this leave a good impression and will bring users back to your PWA. 
+
+## Keep design native-like
+Finally, your PWA experience should feel like a native app so implementing features like Window Controls Overlay can help deliver that experience. You can also use things like Fluent components for Windows based PWAs. Keeping the experience native-like ensures that users have a familiar experience to what they’d expect on a specific platform.
+
+## Resources
+[Fluent design](https://www.microsoft.com/design/fluent/#/)
+
+[Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/)
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/platforms-practices/07.md:44]: Perhaps turn this into a bulleted list of links.

File: docs/30DaysOfPWA/platforms-practices/07.md [hunk 6]
```diff
@@ -1,13 +1,44 @@
...
+
+## Keep content focused and up to date 
+Progressive web apps are inherently about delivering experiences in more performant ways and that experience isn’t just about technical performance but about how quickly users can find what they need. The Starbucks PWA is a great example of focused experience that has been optimized for mobile ordering. The main menu gives us three main options: “Menu”, “Rewards” and “Gift Cards”. 
+
+![The Starbucks PWA home screen and menu](_media/strabucks-app.jpg)
+
+The rest of the site content is deprioritized and placed at the bottom. It’s still accessible but out of the way of the main task users are coming to the site for: to order coffee. 
+
+Identify what key tasks and content your users are coming to your app for and optimize for those scenarios to streamline their task. Focus on reducing any clutter in the UI particularly with mobile versions due to the reduced screen size. For example, the Twitter app on desktop has two sidebars. On the mobile version, the navigation moves to the bottom and the “What’s Happening” sidebar becomes integrated into the search tab with the main focus of the app being the timeline.
```
Comment (captainbrosset) [inline docs/30DaysOfPWA/platforms-practices/07.md:24]: What do you think about adding a screenshot for twitter too?

**Review by captainbrosset — APPROVED** [review]
Looks good to me! I just made a few minor comments.

File: docs/30DaysOfPWA/platforms-practices/07.md (hunk 4 above)
Comment (ststimac) [reply docs/30DaysOfPWA/platforms-practices/07.md:34]: Didn't realize Fluent had a page for it. Adding that.

File: docs/30DaysOfPWA/platforms-practices/07.md (hunk 2 above)
Comment (ststimac) [reply docs/30DaysOfPWA/platforms-practices/07.md:13]: Done

File: docs/30DaysOfPWA/platforms-practices/07.md (hunk 3 above)
Comment (ststimac) [reply docs/30DaysOfPWA/platforms-practices/07.md:28]: Added a link to the offline post from week 1.

**Review by zateutsch — APPROVED** [review]
Awesome!

Thanks, Stephanie.

**Pull request merged**